### PR TITLE
chore: basic test plan document

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Basic document outlining the goals and current state of testing in Forest

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: documentation/developer_documentation/TEST_PLAN.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    Version: 1.0
+    Author: David Himmelstrup
+    Date updated: 2022-11-14
+
+## Test objective:
+
+The Filecoin specification is complex and changes rapidly over time. To manage this complexity, Forest uses a rigorous testing framework, starting with individual functions and ending with complete end-to-end validation. The goals, in descending order of priority, are:
+
+- **Regression detection.** If Forest can no longer connect to main net or if any of its features break, the development team should be automatically notified.
```
Comment (LesnyRumcajs) [inline documentation/developer_documentation/TEST_PLAN.md:11]:
```suggestion
- **Regression detection.** If Forest can no longer connect to mainnet or if any of its features break, the development team should be automatically notified.
```

File: documentation/developer_documentation/TEST_PLAN.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+Bug report template is available on GitHub: https://github.com/ChainSafe/forest/blob/main/.github/ISSUE_TEMPLATE/bug_report.md
+
+The template is applied automatically when bugs are reported through GitHub.
+
+## Risk & Issues:
+
+- We depend on the calibration network for testing. If this network is down, our testing capabilities are degraded.
+- We depend on GitHub Actions for testing. If GitHub Action is unavailable, testing will be degraded.
+- Testing against main net is effective for discovering issues, but not great for identifying root causes. Finding bugs *before* syncing to main net is always to be preferred.
```
Comment (LesnyRumcajs) [inline documentation/developer_documentation/TEST_PLAN.md:66]:
```suggestion
- Testing against mainnet is effective for discovering issues, but not great for identifying root causes. Finding bugs *before* syncing to mainnet is always to be preferred.
```

**Review by LesnyRumcajs — APPROVED** [review]
It would be nice to use the same naming for mainnet as in Filecoin docs, e.g. https://docs.filecoin.io/networks/overview/

**Comment by LesnyRumcajs** [comment]
What is this? Documentation? :scream:

**Review by hanabi1224 — APPROVED** [review]
Should perf test be included here or maybe in a separate doc?

**Comment by lemmih** [comment]
> Should perf test be included here or maybe in a separate doc?

We can update the document once we have a more consistent story for performance testing.

**Pull request merged**